src/Application/Commands/DeleteUnusablePersonRecords.php

```diff
@@ -34,17 +41,78 @@ public function execute(InputInterface $input, OutputInterface $output): int
             new \DateInterval('PT' . Token::COMPLETE_ACCOUNT_VALIDITY_PERIOD_SECONDS . 'S')
         )->format('c');
 
-        $deletedCount = $this->connection->executeStatement(
+        $deletedCount = 0;
+
+        $rows = $this->connection->executeQuery(
             <<<'SQL'
-                    DELETE FROM Person where password is null AND Person.updated_at <= :cutoff
+                    SELECT Person.id, Person.stripe_customer_id FROM Person 
+                    where password is null AND 
+                    Person.updated_at <= :cutoff
                     ORDER BY updated_at
-                    LIMIT 10000; -- we limit to deleting 10 000 at one go to avoid overloading the database.
+                    LIMIT 1000;
```

Comment:
lower limit since each one now has to be deleted individually. Deleting up to 1,000 every hour is still plenty, since there's no backlog of accounts to delete.